src/coreclr/vm/amd64/cgenamd64.cpp

```diff
@@ -206,7 +206,7 @@ void HelperMethodFrame::UpdateRegDisplay(const PREGDISPLAY pRD, bool updateFloat
 
 #endif // TARGET_UNIX
 
-#define CALLEE_SAVED_REGISTER(regname) pRD->pCurrentContextPointers->regname = m_MachState.m_Ptrs.p##regname;
+#define CALLEE_SAVED_REGISTER(regname) pRD->pCurrentContextPointers->regname = (DWORD64 *)(TADDR *)m_MachState.m_Ptrs.p##regname;
```

Comment:
This feels pretty ugly, but `uint64_t` (& `DWORD64`) is defined as `unsigned long long` and `uintptr_t` (& `TADDR`) is defined as `unsigned long`, at least on osx-x64.